src/init/features/project.ts

```diff
@@ -32,48 +34,47 @@ export interface ProjectInfo {
  */
 export async function getProjectInfo(options: any): Promise<ProjectInfo> {
   if (options.project) {
-    return selectProjectFromOptions(options);
+    return toProjectInfo(await getFirebaseProject(options.project));
   }
-  return selectProjectFromList(options);
+  return selectProjectInteractively();
 }
 
-/**
- * Selects project when --project is passed in.
- * @param options Command line options.
- */
-async function selectProjectFromOptions(options: any): Promise<ProjectInfo> {
-  let project: FirebaseProjectMetadata;
-  try {
-    project = await getFirebaseProject(options.project);
-  } catch (e) {
-    throw new FirebaseError(`Error getting project ${options.project}: ${e}`);
+async function selectProjectInteractively(
+  pageSize: number = MAXIMUM_PROMPT_LIST
+): Promise<ProjectInfo> {
+  const { projects, nextPageToken } = await getProjectPage(pageSize);
+  if (projects.length === 0) {
+    throw new FirebaseError("There is no Firebase project associated with this account.");
   }
-  const projectId = project.projectId;
-  const name = project.displayName;
-  return {
-    id: projectId,
-    label: `${projectId} (${name})`,
-    instance: _.get(project, "resources.realtimeDatabaseInstance"),
-    location: _.get(project, "resources.locationId"),
-  };
+  if (nextPageToken) {
+    // Prompt user for project ID if we can't list all projects in 1 page
+    return selectProjectByPrompting();
```

Comment:
I'm not sure I'm a fan of this flow (having to type in a project ID simply because there's many projects).

We have an option in the `init` flow to specify the project to be used: `--project <projectId>`

Always showing the list is predictable behavior and doesn't require typing in a specific string (which isn't a great experience). Also, users could type the incorrect thing; I have a project that is usually displayed as `airportweather-5e16c (AirportWeather)` - both of those seem like reasonable IDs to me, I could see myself picking the wrong one.

This choice I imagine was considered when the original API proposal was made: go/firebase-cli-init-project

This change is something that @mbleigh probably would have to weigh in on, but I think I land on the side of not wanting to change this particular behavior.

Reply:
@bkendall. As far as I remember this behavior was actually [suggested](https://docs.google.com/a/google.com/document/d/1jkwfbu-3H4MNmvKXloMl5SonYxodWuFenZaYLF2_tFo/edit?disco=AAAADDUI_wU) by @mbleigh , due to the fact user may not want to scroll in a list of hundreds of projects + pretty long loading time calling list projects API method (which may take multiple calls due to pagination). It should be lieing 

`--project <projectId>` option is also supported here as well. The CLI will omit the project prompting process in this case.

Reply:
👍